book/src/user_manual/persistence.md

```diff
@@ -4,13 +4,12 @@ The state of a given `MlsGroup` instance is continuously written to the configur
 `StorageProvider`. Later, the `MlsGroup` can be loaded from the provider using
 the `load` constructor, which can be called with the respective storage provider
 as well as the `GroupId` of the group to be loaded. For this to work, the group
-must be stored in the provider.
+must have been written to the provider previously.
 
 ## Forward-Secrecy Considerations
 
-The persisted `MlsGroup` state contains
-private key material. As a consequence, the application needs to delete old group
-states to achieve Forward-Secrecy w.r.t. that key material. Since, as detailed
-above, an old group state is stale immediately after most group operations,
-we recommend that the `StorageProvider` deletes old group state when values are
-overwritten.
+The persisted `MlsGroup` state contains private key material. As a consequence,
+delete old group states need to be deleted in order to achieve Forward-Secrecy
```

Comment:
The `delete` at the beginning of this line is redundant, I think.

Beyond that, I still think this is somewhat confusing. What persisted `MlsGroup` state are we talking about? That persisted (in its individual parts) by the provider? Or is this here in case the application has serialized and (independent of the storage provider) stored the `MlsGroup` state?

Reply:
>The delete at the beginning of this line is redundant, I think.

oops, fixed!

> Beyond that, I still think this is somewhat confusing. What persisted MlsGroup state are we talking about? That persisted (in its individual parts) by the provider? Or is this here in case the application has serialized and (independent of the storage provider) stored the MlsGroup state?

We are talking about the individual pieces of state that are stored in the provider and make up the state of the MlsGroup.